ext/canvas/lib.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::op2;
+use deno_core::ToJsBuffer;
+use image::imageops::FilterType;
+use image::ColorType;
+use image::ImageDecoder;
+use image::Pixel;
+use image::RgbaImage;
+use serde::Deserialize;
+use serde::Serialize;
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum ImageResizeQuality {
+  Pixelated,
+  Low,
+  Medium,
+  High,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct ImageProcessArgs {
+  width: u32,
+  height: u32,
+  surface_width: u32,
+  surface_height: u32,
+  input_x: i64,
+  input_y: i64,
+  output_width: u32,
+  output_height: u32,
+  resize_quality: ImageResizeQuality,
+  flip_y: bool,
+  premultiply: Option<bool>,
+}
+
+#[op2]
+#[serde]
+fn op_image_process(
+  #[buffer] buf: &[u8],
+  #[serde] args: ImageProcessArgs,
+) -> Result<ToJsBuffer, AnyError> {
+  let view =
+    RgbaImage::from_vec(args.width, args.height, buf.to_vec()).unwrap();
```

Comment:
The full allocation here will hurt performance. I think you can create a view into the JS slice:
```js
 let view: ImageBuffer<Rgba<u8>, &[u8]> = ImageBuffer::from_raw(args.witdh, args.height, buf).unwrap();
```

Reply:
sadly I attempted this already, but it wont accept a slice no matter what (even your code, oddly enough, errors out with it wanting a vec), which is weird given that the docs for `from_raw` state that a slice can be passed.